tests/core/instruction/gates/test_gate.py

```diff
@@ -125,3 +126,21 @@ def test_add_product(g1: Gate, g2: Gate, result_matrix: npt.NDArray[np.complex64
 )
 def test_sub_product(g1: Gate, g2: Gate, result_matrix: npt.NDArray[np.complex64]):
     assert np.allclose(g1.minus(g2).to_matrix(), result_matrix)
+
+
+@pytest.mark.parametrize(
+    "gate, targets",
+    [
+        (H.range(4), [0, 1, 2, 3, 4]),
+        (H.range(4, step=2), [0, 2, 4]),
+        (H.range(2, 4), [2, 3, 4]),
```

Comment:
It is said in the documentation that the `end` index is excluded, so we should have `[2,3]`in the targets, no ?

The tests are passing I guess because we iterate over the gates in the circuit, so we stop before addressing the remaining elements in `targets`.

Reply:
fixed in dcd88